routes/v1.js

```diff
@@ -231,6 +301,60 @@ var patchAndValidateTaskDef = function(taskId, taskDef) {
   return null;
 };
 
+/** Ensure the taskGroup exists and that membership is declared */
+let ensureTaskGroup = async (ctx, taskId, taskDef, res) => {
+  let taskGroupId = taskDef.taskGroupId;
+  let taskGroup = await ctx.TaskGroup.load({taskGroupId}, true);
+  let taskGroupExpiration = new Date(
+    new Date(taskDef.expires).getTime() +
+    ctx.taskGroupExpiresExtension * 1000
+  );
+  if (!taskGroup) {
+    taskGroup = await ctx.TaskGroup.create({
+      taskGroupId,
+      schedulerId:  taskDef.schedulerId,
+      expires:      taskGroupExpiration,
+    }).catch(err => {
+      // We only handle cases where the entity already exists
+      if (!err || err.code !== 'EntityAlreadyExists') {
+        throw err;
+      }
+      return ctx.TaskGroup.load({taskGroupId});
```

Comment:
We're not returning `TaskGroup.load()` because we're inside a `.catch(err => {` handler... So really if `TaskGroup.create(...)` fails, `.catch(err => {...` get called and when we return a value from in here that becomes the result of the promise from `TaskGroup.create(...).catch(err => {...` (it's promise chaining... pretty neat..